lib/slack.ml

```diff
@@ -267,13 +256,43 @@ let generate_status_notification (cfg : Config_t.config) (notification : status_
       ]
   in
   let summary =
+    let state_info =
+      match state with
+      | Success -> "succeeded"
+      | Failure -> "failed"
+      | Error -> "error"
+      | _ ->
+        invalid_arg
+          (sprintf "Monorobot doesn't know how to generate notification for the unexpected event %s of %s"
+             (string_of_status_state state) sha
+          )
+    in
+    let commit_message = first_line message in
     match target_url with
     | None ->
-      sprintf "<%s|[%s]> CI Build Status notification: %s" repository.url repository.full_name state_info
+      sprintf "<%s|[%s]>: Build %s for \"%s\"" repository.url repository.full_name state_info commit_message
       (* in case the CI run is not using buildkite *)
     | Some target_url ->
-      sprintf "<%s|[%s]> CI Build Status notification for <%s|%s>: %s" repository.url repository.full_name target_url
-        context state_info
+      (* try to compute the pipeline_url based on the `context` (e.g. buildkite/pipeline) and the `target_url` (link to the build) *)
+      let pipeline_url =
+        let pipeline_name = context |> String.split ~on:'/' |> List.last in
+        match pipeline_name with
+        | None -> None
+        | Some pipeline_name ->
+          let rec loop acc els =
+            match els with
+            | [] -> None
+            | s :: _ when String.(s = pipeline_name) ->
+              let url = pipeline_name :: acc |> List.rev |> String.concat ~sep:"/" in
+              Some url
+            | s :: els -> loop (s :: acc) els
+          in
+          loop [] (String.split ~on:'/' target_url)
+      in
+      ( match pipeline_url with
+      | None -> sprintf "%s: Build %s for \"%s\"" context state_info commit_message
+      | Some pipeline_url -> sprintf "<%s|[%s]>: Build %s for \"%s\"" pipeline_url context state_info commit_message
```

Comment:
this smart behavior should only be applied when we detect that the status notification is from buildkite?

Reply:
also I think that it could be simplified. Check if the notification is from buildkite. If it is, then take the target url and keep it only up to the first 2 elements of the path? Or only the part before `/builds`? No need to be too smart

Reply:
> then take the target url and keep it only up to the first 2 elements of the path?

Done in [`5024756` (#142)](https://github.com/ahrefs/monorobot/pull/142/commits/502475650b620cab2d049a3b828e2f0a3b188135)

> this smart behavior should only be applied when we detect that the status notification is from buildkite?

Isn't that what I did? According to the comment

https://github.com/ahrefs/monorobot/blob/dde2283c2478cd05a648fb3aa6bb4a5a56216f06/lib/slack.ml#L284-L290

My understanding is that if `target_url` is defined (i.e is `Some`) we're in presence of a Buildkite notification

Reply:
It could be a notification from GitHub actions or circle ci or any other kind of CI platform which plugs with GitHub

Reply:
Ok, so I think the comment is misleading. I have changed the logic to detect if this is a buildkite notification. It now checks if the `context` starts with `buildkite` (found nothing better).

I've also changed the text back to what it was originally for the cases where it's not buildkite [`74fe71c` (#142)](https://github.com/ahrefs/monorobot/pull/142/commits/74fe71c3a98a4fd59565287b4dd78332008e906b)

I also adapted the recent code from https://github.com/ahrefs/monorobot/pull/141 to only apply for buildkite notifications